java/vector/src/test/java/org/apache/arrow/vector/ValueVectorDataPopulator.java

```diff
@@ -0,0 +1,379 @@
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(FixedSizeBinaryVector vector, byte[]... values) {
+    final int length = values.length;
+    vector.allocateNewSafe();
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(Float4Vector vector, Float... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(Float8Vector vector, Double... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(IntVector vector, Integer... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(IntervalDayVector vector, IntervalDayHolder... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i].days, values[i].milliseconds);
+      }
+    }
+  }
+
+  public static void setVector(IntervalYearVector vector, Integer... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(SmallIntVector vector, Short... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(TimeMicroVector vector, Long... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(TimeMilliVector vector, Integer... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(TimeNanoVector vector, Long... values) {
```

Comment:
javadoc for in general would be helpful. for all methods (it doesn't need to cover all parameters for obvious ones).  For timestamp where the value has specific semantics it would be good to document what it represents (e.g. nanonseconds since the unix epoch)